atlasdb-commons/src/test/java/com/palantir/common/proxy/PredicateSwitchedProxyTest.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.common.proxy;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.junit.Test;
+
+public class PredicateSwitchedProxyTest {
+    private final TestInterface decorated = mock(TestInterface.class);
+    private final TestInterface delegate = mock(TestInterface.class);
+
+    private final AtomicBoolean atomicBoolean = new AtomicBoolean(false);
+
+    private final TestInterface testInterface = PredicateSwitchedProxy.newProxyInstance(
+            decorated,
+            delegate,
+            atomicBoolean::get,
+            TestInterface.class);
+
+    @Test
+    public void dynamicallySwitchesCorrectlyForNoArgMethods() throws Exception {
```

Comment:
can this be nuked? in favour of the test below

Reply:
yeah, this is a leftover from when this was used to test TimestampService I think